### ISSUE TITLE
TINYINT is not supported by table inspector currently

### ISSUE THREAD

**Issue by ThePlenkov** [SAP-samples/hana-developer-cli-tool-example#27]
Hi! inspectTables doesn't support TINYINT currently:

![image](https://user-images.githubusercontent.com/6381507/112467854-d8993800-8d67-11eb-9fa1-1f3b6a604cdc.png)

Will provide PR now for this issue

**Comment by jung-thomas** [comment]
Actually I intentionally don't map data types that don't have direct correlation to a CDS type .  See the Predefined Types section of this document:
https://cap.cloud.sap/docs/cds/cdl

This way the person converting can make the choice if they want to either use a non-exact match or use a HANA-Specific data type. However I think that needs to be a decision that the person converting needs to make.

**Comment by ThePlenkov** [comment]
hm.. but then it just doesn't fit automated scenarios
In our pipeline schemas for cross-schema reference are loaded automatically before build.
If we have HANA database behind ( and this tool is for hana entirely)  i guess it's ok if we also support native hana types in CDS

**Comment by ThePlenkov** [comment]
> Actually I intentionally don't map data types that don't have direct correlation to a CDS type . See the Predefined Types section of this document:
> https://cap.cloud.sap/docs/cds/cdl
> 
> This way the person converting can make the choice if they want to either use a non-exact match or use a HANA-Specific data type. However I think that needs to be a decision that the person converting needs to make.

Yes, but from the other side. Those types are uniquely point to hana.* cds types by their name as they are not in the common list:
![image](https://user-images.githubusercontent.com/6381507/112474306-c0c5b200-8d6f-11eb-850c-212ddb124311.png)

**Comment by jung-thomas** [comment]
I like your idea of using an optional parameter to support the hana specific types.  We don't want to encourage automatic usage of such types as they are really discouraged and do break the CAP database independence model.  However if someone wants to use an automated migration approach as you describe this lets them make the decision en masse.

**Comment by ThePlenkov** [comment]
Just tested locally - it works:
`hana-cli massConvert SFLIGHT --hana`

**Comment by ThePlenkov** [comment]
PR is ready for review - please check it. I have already used it locally to resolve conflicts - everything works as expected, also backward compatible

**Comment by jung-thomas** [comment]
Accepted pull request but also made some adjustments afterwards.  The fall through logic didn't work like expected. If you don't break within a switch block it automatically executes the next switch block without checking the condition. This lead to some incorrect types when you don't use the new HANA native types feature. 

I also extended the new feature to inspectView and cds commands as well.

I added logic to also support the two HANA geo types that have matches in CDS (including a lookup of the srid from ST_GEOMETRY_COLUMNS. 

I'll push the new version to NPM momentarily.
